gateway/ln-gateway/src/gatewayd/actor.rs

```diff
@@ -33,6 +33,12 @@ pub struct GatewayActor {
     task_group: TaskGroup,
 }
 
+#[derive(Debug, Clone)]
+pub enum BuyPreimage {
```

Comment:
when we use this struct, have we not already bought the preimage? so `BoughtPreimage`?

Reply:
Id want to get rid of the struct all together. `BuyPreimage::Internal` should not be a part of gateways functionality, so we really don't need this enum.

Issue #613 is relevant

Reply:
thanks for the reference